idaes/generic_models/unit_models/tests/test_heat_exchanger_lc.py

```diff
@@ -0,0 +1,515 @@
+        assert pytest.approx(0.0, abs=1e-3) == value(
+                sapon.fs.unit.outlet_1.conc_mol_comp[0, "SodiumAcetate"])
+        assert pytest.approx(0.0, abs=1e-3) == value(
+                sapon.fs.unit.outlet_1.conc_mol_comp[0, "Ethanol"])
+
+        assert pytest.approx(55388.0, rel=1e-3) == value(
+                sapon.fs.unit.outlet_2.conc_mol_comp[0, "H2O"])
+        assert pytest.approx(100.0, rel=1e-3) == value(
+                sapon.fs.unit.outlet_2.conc_mol_comp[0, "NaOH"])
+        assert pytest.approx(100.0, rel=1e-3) == value(
+                sapon.fs.unit.outlet_2.conc_mol_comp[0, "EthylAcetate"])
+        assert pytest.approx(0.0, abs=1e-3) == value(
+                sapon.fs.unit.outlet_2.conc_mol_comp[0, "SodiumAcetate"])
+        assert pytest.approx(0.0, abs=1e-3) == value(
+                sapon.fs.unit.outlet_2.conc_mol_comp[0, "Ethanol"])
+
+        assert pytest.approx(301.3, abs=1e-1) == \
+            value(sapon.fs.unit.outlet_1.temperature[0])
+        assert pytest.approx(318.7, abs=1e-1) == \
+            value(sapon.fs.unit.outlet_2.temperature[0])
+
+        assert pytest.approx(101325, abs=1e2) == \
+            value(sapon.fs.unit.outlet_1.pressure[0])
+        assert pytest.approx(101325, abs=1e2) == \
+            value(sapon.fs.unit.outlet_2.pressure[0])
+
+
+class TestHXLCGeneric(object):
+
+    @pytest.fixture()
+    def unconstrained_model(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": True,
+                                       "time_set": [0, 1],
+                                       "time_units": pyunits.s})
+
+        m.fs.properties = iapws95.Iapws95ParameterBlock()
+
+        m.fs.unit = HeatExchangerLumpedCapacitance(default={
+            "shell": {"property_package": m.fs.properties},
+            "tube": {"property_package": m.fs.properties},
+            "dynamic": False})
+
+        return m
+
+    @pytest.fixture()
+    def model(self, unconstrained_model):
+
+        m = unconstrained_model
+        m.discretizer = TransformationFactory('dae.finite_difference')
+        m.discretizer.apply_to(m, nfe=1, wrt=m.fs.time, scheme="BACKWARD")
+
+        m.fs.unit.area.fix(1000)
+        m.fs.unit.heat_capacity = 100
+
+        m.fs.unit.shell_inlet.flow_mol[:].fix(100)
+        m.fs.unit.shell_inlet.pressure[:].fix(101325)
+        m.fs.unit.shell_inlet.enth_mol[:].fix(4000)
+
+        m.fs.unit.tube_inlet.flow_mol[:].fix(100)
+        m.fs.unit.tube_inlet.pressure[:].fix(101325)
+        m.fs.unit.tube_inlet.enth_mol[:].fix(3000)
+
+        m.fs.unit.ua_cold_side[:].fix(100)
+        m.fs.unit.ua_hot_side[:].fix(100)
+
+        return m
+
+    @pytest.mark.unit
+    def test_units(self, model):
+        # Note: using the discretizer makes the units of measure on the time
+        # derivative term inconsistent...
+        with pytest.raises(InconsistentUnitsError):
+            assert_units_consistent(model)

```

Comment:
Rather than expecting an error here, I think the correct approach would be to `xfail` the test (i.e. tell pytest that for now it is expected to fail). The difference is rather semantic, but in this case to me an `xfail` tends to indicate this is wrong and needs to be fixed but we will accept it for now, whilst the `pytest.rasies` tends to imply an expected behaviour we are testing for.